rss2email/config.py

```diff
@@ -61,6 +83,36 @@ def setup_html2text(self, section='DEFAULT'):
             section, 'links-after-each-paragraph')
         _html2text.BODY_WIDTH = self.getint(section, 'body-width')
 
+    def set_overrides(self, overrides):
+        self.overrides = overrides
+
+    def read(self, filenames):
+        # if we read in from a file, we need to keep track of the
+        # filenames that produced this config for later comparison, so
+        # we don't overwrite disk configuration with command-line
+        # overrides except in some specific cases
+        read_filenames = super(Config, self).read(filenames)
+        self.filenames = read_filenames
+        return read_filenames
+
+    def write(self, f):
+        """Writes config to file. Overrides are writen to disk only if the
+        section being overridden does not exist on disk (e.g. when
+        creating a new feed)"""
```

Comment:
I’m not sure this is a good way to handle the overrides; because it will sometimes modify the on-disk config and sometimes not, and there is no clean way for the user to see when that really happens and when not. See also the other comments.

Reply:
The whole point of this feature is to enable users to set config keys on the command line to be written to disk when adding a new feed, and that is the only time overrides are written to disk, since that is the only time the program adds a new section.

Reply:
I guess just putting it in the `add` command would help :)